src/Admin/BaseBlockAdmin.php

```diff
@@ -116,7 +130,14 @@ public function postUpdate($object)
      */
     public function prePersist($object)
     {
-        $this->blockManager->get($object)->prePersist($object);
+        if (\is_callable([$object, 'prePersist'])) {
+            $this->blockManager->get($object)->prePersist($object);
+
+            @trigger_error(
+                'The '.__METHOD__.' method is deprecated since version 3.x and will be removed in sonata/block-bundle 4.0.',
```

Comment:
```suggestion
                'The '.__METHOD__.'() method is deprecated since sonata-project/block-bundle 3.x and will be removed in version 4.0.',
```